client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java

```diff
@@ -133,6 +134,21 @@ public class DefaultMQProducer extends ClientConfig implements MQProducer {
      */
     private TraceDispatcher traceDispatcher = null;
 
+    /**
+     * Indicate whether to block message when asynchronous sending traffic is too heavy.
+     */
+    private boolean enableBackpressureForAsyncMode = false;
+
+    /**
+     * on BackpressureForAsyncMode, limit maximum number of on-going sending async messages
+     */
+    private Semaphore semaphoreAsyncNum = new Semaphore(60000, true);
+
```

Comment:
expose the integer value instead of Semaphore directly.

It is better to initialize the semaphore in the constructor of DefaultMQProducerImpl

Reply:
default to 10000 is enough.

Reply:
It's a good suggestion. I have fixed it.